test/test_reload.py

```diff
@@ -0,0 +1,62 @@
+#!/usr/bin/env python3
+# -*- coding: utf-8 -*-
+
+servers_list = [13301, 13302]
+
+
+def test_reload_slow(servers, helpers):
+    """ Check that hot-reload doesn't affect statuses """
+
+    assert servers[13301].probe_uri('localhost:13302') is True
+    assert servers[13302].get_member('localhost:13301')['status'] == 'alive'
+
+    servers[13302].conn.eval('''
+        local log = require('log')
+        local yaml = require('yaml')
+        local fiber = require('fiber')
+
+        _G.guard = fiber.new(function()
+            membership.subscribe():wait()
+            fiber.testcancel()
+            log.error('Unexpected event:')
+            log.error(yaml.encode(membership.members()))
+            os.exit(1)
+        end)
+    ''')
+
+    assert servers[13301].conn.eval('''
+        local log = require('log')
+        local fiber = require('fiber')
+
+        package.loaded['membership'] = nil
+        log.info('Membership unloaded')
+        fiber.sleep(1)
+
+        _G.membership = require('membership')
+        log.info('Membership reloaded')
+        fiber.sleep(1)
+
+        log.info('Doing file %s...', arg[0])
+        dofile(arg[0])
+        log.info('Dofile succeeded')
+        fiber.sleep(1)
+
+        return membership.probe_uri('localhost:13302')
+    ''') == [True]
+
+    servers[13302].conn.eval('''
+        _G.guard:cancel()
+    ''')
+
+
+def test_reload_fast(servers, helpers):
```

Comment:
Please, add a test where you change some "option" and its value preserved (or not?) after reload